astronomer/providers/amazon/aws/sensors/s3.py

```diff
@@ -89,35 +90,39 @@ def execute(self, context: Context) -> None:
             else:
                 raise e
         if not poke:
-            self.defer(
-                timeout=timedelta(seconds=self.timeout),
-                trigger=S3KeyTrigger(
-                    bucket_name=cast(str, self.bucket_name),
-                    bucket_key=self.bucket_key,
-                    wildcard_match=self.wildcard_match,
-                    check_fn=self.check_fn,
-                    aws_conn_id=self.aws_conn_id,
-                    verify=self.verify,
-                    poke_interval=self.poke_interval,
-                    soft_fail=self.soft_fail,
-                ),
-                method_name="execute_complete",
-            )
+            self._defer()
+
+    def _defer(self) -> None:
+        self.defer(
+            timeout=timedelta(seconds=self.timeout),
+            trigger=S3KeyTrigger(
+                bucket_name=cast(str, self.bucket_name),
+                bucket_key=self.bucket_key,
+                wildcard_match=self.wildcard_match,
+                aws_conn_id=self.aws_conn_id,
+                verify=self.verify,
+                poke_interval=self.poke_interval,
+                soft_fail=self.soft_fail,
+                should_check=self.should_check,
```

Comment:
What happens if we pass the `check_fn` into `S3KeyTrigger`? I thought we should run this inside `S3KeyTrigger` instead of `execute_complete`